shark_turbine/kernel/wave/codegen.py

```diff
@@ -231,6 +242,8 @@ def gen_sympy_index(emitter: WaveEmitter, expr: sympy.Expr) -> OpResult:
                 # But check whether floordivsi is needed.
                 stack.append(stack.pop())
             case sympy.Rational():
+                # `x * (a/b)` transformed into `(x * a) / b`
+                # `x + (a/b)` transformed into `(x*b + a) / b`
```

Comment:
Why do we need to transform x + (a/b) - > (x*b + a) / b? Can't we do (a/b) and then add x? Just trying to avoid an extra multiplication.

Reply:
After all the sympy expansions I ended with expressions like `floor(x + 3/4)` which will be rounded to `x` if we do division first. I'm still not sure if any of this correct, but I followed the existing `mul` approach.

Reply:
I see. So a and b are integers and / corresponds to signed integer division, so the only way to ensure the correct result is to do the multiplication first (and addition) and then the division. I think this is a reasonable approach to take considering the alternative which would involve casting a, b to floats and doing all the arithmetic on floats.